ckanext/georchestra/utils/ldap_utils.py

```diff
@@ -0,0 +1,303 @@
+            pages += 1
+            # Send search request
+            try:
+                response = cnx.search_ext(dn,
+                                      ldap.SCOPE_ONELEVEL,
+                                      filter,
+                                      attrlist=attributes,
+                                      serverctrls=[page_control])
+            except ldap.LDAPError as e:
+                log.error('LDAP search failed: %s' % e)
+
+            # Pull the results from the search request
+            try:
+                rtype, rdata, rmsgid, serverctrls = cnx.result3(response)
+            except ldap.LDAPError as e:
+                log.error('Could not pull LDAP results: %s' % e)
+
+            # Each "rdata" is a tuple of the form (dn, attrs), where dn is
+            # a string containing the DN (distinguished name) of the entry,
+            # and attrs is a dictionary containing the attributes associated
+            # with the entry. The keys of attrs are strings, and the associated
+            # values are lists of strings.
+            for item in rdata:
+                processed_id = process_item(item, external_process, context)
+                processed_items.append(processed_id)
+
+            # Get cookie for next request
+            result.extend(rdata)
+            controls = [control for control in serverctrls
+                        if control.controlType == SimplePagedResultsControl.controlType]
+            if not controls:
+                print('The server ignores RFC 2696 control')
+                break
+
+            # Ok, we did find the page control, yank the cookie from it and
+            # insert it into the control for our next search. If however there
+            # is no cookie, we are done!
+
+            page_control.cookie = controls[0].cookie
+            if not controls[0].cookie:
+                break
+        return processed_items
+
+    def _process_org(self, org, external_process, context):
+        """
+        Applied to every organization entry listed during the LDAP scan
+        :param org:
+        :param external_process:
+        :param context:
+        :return:
+        """
+        org = self._org_format_and_complete(org)
+        external_process(context, org)
+        return org['id']
+
+    def _process_user(self, user, external_process, context):
+        """
+        Applied to every user entry listed during the LDAP scan
+        :param user:
+        :param external_process:
+        :param context:
+        :return:
+        """
+        # filter out nosync users like geoserver_privileged_user
+        if user[1]['uid'][0] in self.nosync_users_list:
+            return None
+        user = self._user_format_and_complete(user)
+        external_process(context, user)
+        return user['id']
+
+    def _org_format_and_complete(self, org):
+        """
+        Gets complementary attributes from LDAP (organization information is split into 3 objects)
```

Comment:
Fortunately, only 2 !

Reply:
Yes, and hopefully 1 in close future ?

Reply:
Not sure about it. Nothing planned for now.

Reply:
Ah ? OK